packages/token-bridge-sdk/src/hooks/useArbTokenBridge.ts

```diff
@@ -154,60 +160,96 @@ export const useArbTokenBridge = (
   const depositEth = async ({
     amount,
     l1Signer,
-    txLifecycle
+    txLifecycle,
+    destinationAddress
   }: {
     amount: BigNumber
     l1Signer: Signer
-    txLifecycle?: L1EthDepositTransactionLifecycle
+    txLifecycle?:
+      | L1EthDepositTransactionLifecycle & L1ContractCallTransactionLifecycle
+    destinationAddress?: string
   }) => {
     const ethBridger = await EthBridger.fromProvider(l2.provider)
 
-    let tx: L1EthDepositTransaction
+    verifyDestinationAddressOrThrow({
+      provider: l1Signer.provider,
+      destinationAddress,
+      isDeposit: true
+    })
 
-    try {
-      tx = await ethBridger.deposit({
-        amount,
-        l1Signer
+    let tx: L1EthDepositTransaction & L1ContractCallTransaction
+    let receipt:
+      | L1EthDepositTransactionReceipt
+      | L1ContractCallTransactionReceipt
+
+    const _addTransaction = (txHash: string) =>
```

Comment:
why abstract it out when it's only used once?

Reply:
was for my previous implementation, will remove